server/publications/room.js

```diff
@@ -112,6 +112,13 @@ Meteor.methods({
 	},
 });
 
+const getSubscriptions = (clientAction, id) => {
+	const fields = { 'u._id': 1 };
+	return clientAction === 'removed'
+		? RocketChat.models.Subscriptions.trashFind({ rid: id }, { fields })
+		: RocketChat.models.Subscriptions.findByRoomId(id, { fields });
```

Comment:
This is a problem! Why get all users of a room when it changes? A room changes for every message and reaction.

The deletion part I understand and that will happens only one time. But what is the objective with the rest?